pkg/traceql/storage.go

```diff
@@ -68,12 +89,18 @@ type FetchSpansRequest struct {
 
 	// Hints
 
-	// By default the storage layer fetches spans meeting any of the criteria.
+	// By default, the storage layer fetches spans meeting any of the criteria.
 	// This hint is for common cases like { x && y && z } where the storage layer
 	// can make extra optimizations by returning only spansets that meet
 	// all criteria.
 	AllConditions bool
 
+	// Exemplars hint tells the storage layer that the query wants to collect exemplars.
+	// This allows to optimise second pass iterators to quit if we have enough samples.
+	// Iterating over all spans to collect exemplars is very expensive,
+	// specially in allocs and mem.
+	Exemplars bool
```

Comment:
I think this is too specialized, and also lacks a way to specify the max number of exemplars. Would prefer a way to expose the max count, or a callback that stops the iterator early that query_range could use when it has enough exemplars.